sci-libs/vtk/vtk-7.0.0.ebuild

```diff
@@ -0,0 +1,362 @@
+		|| die
+
+	cmake-utils_src_prepare
+}
+
+src_configure() {
+	# general configuration
+	local mycmakeargs=(
+		-Wno-dev
+#		-DVTK_DIR="${S}"
+		-DVTK_INSTALL_LIBRARY_DIR=$(get_libdir)
+#		-DVTK_DATA_ROOT="${EPREFIX}/usr/share/${PN}/data"
+		-DVTK_CUSTOM_LIBRARY_SUFFIX=""
+		-DBUILD_SHARED_LIBS=ON
+		-DVTK_USE_SYSTEM_AUTOBAHN=ON
+		-DVTK_USE_SYSTEM_EXPAT=ON
+		-DVTK_USE_SYSTEM_FREETYPE=ON
+		-DVTK_USE_SYSTEM_FreeType=ON
+		-DVTK_USE_SYSTEM_GL2PS=ON
+		-DVTK_USE_SYSTEM_HDF5=ON
+		-DVTK_USE_SYSTEM_JPEG=ON
+		-DVTK_USE_SYSTEM_LIBPROJ4=OFF
+		-DVTK_USE_SYSTEM_LIBXML2=ON
+		-DVTK_USE_SYSTEM_LibXml2=ON
+		-DVTK_USE_SYSTEM_NETCDF=ON
+		-DVTK_USE_SYSTEM_OGGTHEORA=ON
+		-DVTK_USE_SYSTEM_PNG=ON
+		-DVTK_USE_SYSTEM_TIFF=ON
+		-DVTK_USE_SYSTEM_TWISTED=ON
+		-DVTK_USE_SYSTEM_XDMF2=OFF
+		-DVTK_USE_SYSTEM_XDMF3=OFF
+		-DVTK_USE_SYSTEM_ZLIB=ON
+		-DVTK_USE_SYSTEM_ZOPE=ON
+		-DVTK_USE_SYSTEM_LIBRARIES=ON
+		-DVTK_USE_GL2PS=ON
+		-DVTK_USE_LARGE_DATA=ON
+		-DVTK_USE_PARALLEL=ON
+		-DVTK_EXTRA_COMPILER_WARNINGS=ON
+		-DVTK_Group_StandAlone=ON
+	)
+
+	mycmakeargs+=(
+		-DBUILD_DOCUMENTATION=$(usex doc)
+		-DBUILD_EXAMPLES=$(usex examples)
+#		-DBUILD_VTK_BUILD_ALL_MODULES_FOR_TESTS=$(usex test)
+		-DVTK_BUILD_ALL_MODULES=$(usex all-modules)
+#		-DUSE_DOCUMENTATION_HTML_HELP=$(usex doc)
+		-DVTK_Group_Imaging=$(usex imaging)
+		-DVTK_Group_MPI=$(usex mpi)
+		-DVTK_Group_Rendering=$(usex rendering)
+		-DVTK_Group_Tk=$(usex tk)
+		-DVTK_Group_Views=$(usex views)
+		-DVTK_Group_Web=$(usex web)
+		-DVTK_WWW_DIR="${ED%/}/${MY_HTDOCSDIR}"
+		-DVTK_WRAP_JAVA=$(usex java)
+		-DVTK_WRAP_PYTHON=$(usex python)
+		-DVTK_WRAP_PYTHON_SIP=$(usex python)
+		-DVTK_WRAP_TCL=$(usex tcl)
+		-DVTK_USE_BOOST=$(usex boost)
+		-DUSE_VTK_USE_BOOST=$(usex boost)
+		-DModule_vtkInfovisBoost=$(usex boost)
+		-DModule_vtkInfovisBoostGraphAlgorithms=$(usex boost)
+#		-DVTK_USE_CG_SHADERS=$(usex cg)
+		-DVTK_USE_ODBC=$(usex odbc)
+		-DModule_vtkIOODBC=$(usex odbc)
+		-DVTK_USE_OFFSCREEN=$(usex offscreen)
+		-DVTK_OPENGL_HAS_OSMESA=$(usex offscreen)
+#		-DvtkFiltersSMP=$(usex smp)
+		-DVTK_USE_OGGTHEORA_ENCODER=$(usex theora)
+		-DVTK_USE_NVCONTROL=$(usex video_cards_nvidia)
+		-DModule_vtkFiltersStatisticsGnuR=$(usex R)
+		-DVTK_USE_X=$(usex X)
+	# IO
+		-DVTK_USE_FFMPEG_ENCODER=$(usex ffmpeg)
+		-DModule_vtkIOGDAL=$(usex gdal)
+		-DModule_vtkIOGeoJSON=$(usex json)
+		-DModule_vtkIOXdmf2=$(usex xdmf2)
+	# Apple stuff, does it really work?
+		-DVTK_USE_COCOA=$(usex aqua)
+	)
+
+	if use examples || use test; then
+		mycmakeargs+=( -DBUILD_TESTING=ON )
+	fi
+
+	if use kaapi; then
+		mycmakeargs+=( -DVTK_SMP_IMPLEMENTATION_TYPE="Kaapi" )
+	elif use tbb; then
+		mycmakeargs+=( -DVTK_SMP_IMPLEMENTATION_TYPE="TBB" )
+	else
+		mycmakeargs+=( -DVTK_SMP_IMPLEMENTATION_TYPE="Sequential" )
+	fi
+
+	if use java; then
+		if [[ ${ARCH} == amd64 ]]; then
+			mycmakeargs+=( -DJAVA_AWT_LIBRARY="${JAVA_HOME}/jre/lib/${ARCH}/libjawt.so;${JAVA_HOME}/jre/lib/${ARCH}/xawt/libmawt.so" )
```

Comment:
As noted in [bug #552796](https://bugs.gentoo.org/show_bug.cgi?id=552796), this whole `if use java` block should just go away. I should have addressed this sooner but this is the first time I've looked at it long enough to understand what's going on.